qualcomm.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import "net"
+
+const (
+	// HPAV EtherType
+	etherType = 0x88e1
+
+	hpVersion = 0
+
+	nwInfoReq = 0xa038
+	nwInfoCnf = 0xa039
+)
+
+var (
+	hpVendor = [...]byte{0x00, 0xb0, 0x52}
+)
+
+// macAddr is a convenience type that is strictly limited to the Ethernet MAC address length,
+// unlike the standard library net.HardwareAddr type.
+type macAddr [6]byte
+
+func (m macAddr) String() string {
+	return net.HardwareAddr(m[:]).String()
+}
+
+type networkID [7]byte
+
+func (n networkID) String() string {
+	return net.HardwareAddr(n[:]).String()
+}
```

Comment:
These types and functions are not Qualcomm-specific, wouldn't it make more sense to place them in `macaddress.go`?

Reply:
I was thinking that the current `macAddressValue` type (which is an alias for `net.HardwareAddr`) should perhaps be renamed to `probeAddress`, since its main job is to take a non-numeric address "enum" as a command-line flag, and provide the associated numeric address. The "local" MAC alias (`00:b0:52:00:00:01`) is Qualcomm Atheros specific, so we definitely have a little bit of sorting out to do.

I could possibly merge the functionality of the existing `macAddressValue` type into this new `macAddr` type. We don't really need to type-alias `net.HardwareAddr` - the value only needs to typecast in the `write_homeplug()` function, i.e. for the mdlayher/ethernet functions to use. Everything up to that point is strictly 6-byte Ethernet MAC addresses, and does not need the extra baggage of Go's more generic `net.HardwareAddr`.

Reply:
Bear in mind that the code in that file was probably copy-pasted from examples I found somewhere, the only reason for it was to allow passing addresses in a more flexible way (reusing the `net.HardwareAddr` parser) and the helper aliases.